### ISSUE TITLE
Issues with YOLO pose training - precision and recall to zero

### ISSUE THREAD

**Issue by peppenap98** [ultralytics/ultralytics#22104]
Hello, I am trying to train YOLOv11n for a pose task on a dataset. However, while object detection metrics rapidly improve, pose precision and recall metrics always remain exactly equal to zero. 

To exclude problems with my dataset, I tried to use the baseline commands reported in https://docs.ultralytics.com/it/tasks/pose/#train, reproducing a 100 epochs training on coco8 dataset. In this case, pose-related metrics rapidly decrease and finally become equal to 0, as in the image below.

<img width="1281" height="444" alt="Image" src="https://github.com/user-attachments/assets/42db814e-452f-43d7-9d92-3214eddb64b0" />

I had already used YOLO11 and v8 for pose tasks before, and I had never got metrics which were exactly equal to 0 and which never increased. It seems also weird that it also occurs with the basic lines reported in the link. Is anyone facing a similar issue?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @peppenap98, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will join the thread soon to assist further. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. For your pose task with YOLO11n showing zero precision/recall, please include:
-Exact CLI/Python commands used (for both your dataset and the coco8 pose baseline), model variant, and any custom args
-Full training and validation logs, including metrics and any warnings
-ultralytics and torch versions, OS, Python version, GPU model, CUDA/CUDNN info
-A small sample of images+labels (e.g., 1–2 train/val files) to verify keypoint formats
-The data.yaml used and any changes to default settings
-Screenshots/plots like your metrics chart 📈

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
What's the training command you're using?

COCO8 is a toy dataset. It's not supposed to give a good performance in training. It just trains on 4 images.

**Comment by srcejon** [comment]
>  Is anyone facing a similar issue?

Yes.

**Comment by srcejon** [comment]
> What's the training command you're using?
> 
> COCO8 is a toy dataset. It's not supposed to give a good performance in training. It just trains on 4 images.

I'm having a similar problem with a larger custom dataset, so I thought I'd try COCO8 as known reference point and noticed it doing similar. 

Just using the example commands from the documentation:

    from ultralytics import YOLO

    if __name__ == '__main__': # fix for windows
        model = YOLO("yolo11l-pose.pt")  # load a pretrained model (recommended for training)
        results = model.train(data="coco8-pose.yaml", epochs=100, imgsz=640)

l model gives:

<img width="4000" height="1200" alt="Image" src="https://github.com/user-attachments/assets/d4e444cb-e3c8-4508-86fe-aff711a60849" />

s model:

<img width="4000" height="1200" alt="Image" src="https://github.com/user-attachments/assets/264e8a31-1047-462c-a4c0-7201bbd266f7" />

n model:

<img width="4000" height="1200" alt="Image" src="https://github.com/user-attachments/assets/a8892e99-cd14-4734-9a8b-d9d4063c6968" />

**Comment by srcejon** [comment]
Adding in extra training images from the full coco-pose dataset does help:

47 images:

<img width="4000" height="1200" alt="Image" src="https://github.com/user-attachments/assets/0722d99b-377f-4d2a-85c6-1eda4db3cb2e" />

85 images:

<img width="4000" height="1200" alt="Image" src="https://github.com/user-attachments/assets/9bf0860a-9f43-4e1c-b1fc-390ceeb886c9" />

**Comment by glenn-jocher** [comment]
Thanks for the detailed plots — your observation matches expectations: COCO8‑Pose (4 train/4 val) is only for pipeline sanity checks, and with so few people OKS-based pose metrics can read as zero; the improvement you saw when adding images confirms this behavior. For a quick environment sanity check, please evaluate an official pose model on the full COCO‑Pose val split; you should see non‑zero pose metrics:

```bash
yolo pose val model=yolo11n-pose.pt data=coco-pose.yaml device=0
```

```python
from ultralytics import YOLO
YOLO("yolo11n-pose.pt").val(data="coco-pose.yaml", device=0)
```

If you still get zeros, please paste the `yolo checks` output, share the text `results.csv` (not screenshots), and confirm your device; if you’re on Apple MPS, run with `device=cpu` due to a known pose issue noted in the [PoseTrainer reference](https://docs.ultralytics.com/reference/models/yolo/pose/train/). For custom data, verify your dataset YAML includes the correct `kpt_shape` and `flip_idx` like the example on the [COCO8‑Pose dataset page](https://docs.ultralytics.com/datasets/pose/coco8-pose/); for benchmarking, use the full [COCO‑Pose dataset](https://docs.ultralytics.com/datasets/pose/coco/).
